FlashApp.sh

```diff
@@ -0,0 +1,247 @@
+    umount $ota_rootfs_mount_point
+    exit 1
+else
+    # We have two rootfs partitions; one is the current rootfs and the other is the new rootfs.
+    # identify the current rootfs partition and the new rootfs partition.
+    activeBankDev=$(sed -e "s/.*root=//g" /proc/cmdline | cut -d ' ' -f1)
+    if [ "$activeBankDev" == "/dev/mmcblk0p2" ]; then
+        passiveBankDev="/dev/mmcblk0p3"
+    else
+        passiveBankDev="/dev/mmcblk0p2"
+    fi
+    logger "Active rootfs partition: '$activeBankDev' and Passive partition: '$passiveBankDev'."
+    logger "Updating '$passiveBankDev' with the contents of '$ota_rootfs_mount_point'"
+
+    dontUmountPassiveBank=0
+    # check if the passive rootfs partition is already mounted; if so, use it.
+    # TODO: handle RO mounted partitions.
+    passiveBankMountPoint=$(mount | grep $passiveBankDev | awk '{print $3}')
+    if [ -n "$passiveBankMountPoint" ]; then
+        dontUmountPassiveBank=1
+        logger "Passive rootfs partition '$passiveBankDev' is already mounted at '$passiveBankMountPoint'; proceeding with that."
+        target_rootfs_mount_point=$passiveBankMountPoint
+        logger "Using premounted target_rootfs_mount_point: $target_rootfs_mount_point"
+    else
+        mount -t ext4 $passiveBankDev $target_rootfs_mount_point
+        if [ $? -ne 0 ]; then
+            logger "Failed to mount the passive rootfs partition '$passiveBankDev'; cannot proceed, exiting."
+            umount $ota_rootfs_mount_point
+            # Try unmounting target_rootfs_mount_point and exit; may fail.
+            umount $target_rootfs_mount_point
+            exit 1
+        fi
+    fi
+    logger "Copying the contents of '$ota_rootfs_mount_point' to '$target_rootfs_mount_point'"
+    rm -rf $target_rootfs_mount_point/* && sync
+    cp -ar $ota_rootfs_mount_point/* $target_rootfs_mount_point/ && sync
+    if [ $? -ne 0 ]; then
+        logger "Failed to copy the contents of '$ota_rootfs_mount_point' to '$target_rootfs_mount_point'; revert and abort."
+    else
+        logger "The contents of '$ota_rootfs_mount_point' are copied to '$target_rootfs_mount_point' successfully."
+        echo "OTA_UPDATED_ROOTFS="$(date)"" >> $target_rootfs_mount_point/version.txt
+        isRootFSUpdateSuccess=1
+    fi
+    if [ $dontUmountPassiveBank -eq 0 ]; then
+        umount $target_rootfs_mount_point
+    fi
+    umount $ota_rootfs_mount_point
+fi
+
+isBootUpdateSuccess=0
+
+# mount the P1 partition which is Boot partition as readonly to avoid any accidental writes
+ota_boot_node=$losetupNode"p1"
+logger "Mounting the rootfs partition '$ota_boot_node' at '$ota_boot_mount_point' as read-only"
+mount -o ro $ota_boot_node $ota_boot_mount_point
+if [ $? -ne 0 ]; then
+    logger "Failed to mount $ota_boot_node at $ota_boot_mount_point; exiting."
+    exit 1
+else
+    logger "Copying the contents of '$ota_boot_mount_point' to '/boot'"
+    cp -ar $ota_boot_mount_point/* /boot/ && sync
```

Comment:
This is not a clean copy. Whatever was present extra in /boot/ will not get removed. Is there a better way

Reply:
Updated logic to clean it.